### DISCUSSION TITLE
How we can make it "multi" thread?

### DISCUSSION THREAD

**Discussion by ch3shere** [innightwolfsleep/llm_telegram_bot#49] [General]
I mean it can handle only 1 request? or simultaneously it's answering each user independently?

**Comment by innightwolfsleep** [comment]
At the moment, telegram_bot handles only one request simultaneously .
In my practice, parallel works overflows VRAM, so i refused multithreading during answer generating.

If multithreading generator is requested by users, further i can add it.

**Reply by ch3shere** [reply]
> At the moment, telegram_bot handles only one request simultaneously . In my practice, parallel works overflows VRAM, so i refused multithreading during answer generating.
> 
> If multithreading generator is requested by users, further i can add it.

Yeah, i agree that multi threading overflows VRAM. But let's assume if TG bot will be used by 10 members, won't it confuse "prompts"?

**Reply by ch3shere** [reply]
And also is it possible to support siletto?
`"<audio src="file/extensions/silero_tts/outputs/Alpaca_1685347368.wav" controls autoplay></audio>"`

**Reply by innightwolfsleep** [reply]
Each user have own history. No prompts shuffling betwen them.

**Reply by ch3shere** [reply]
Thanks bruh! And what's about silero tts answers being forwarded as voice message (and also maybe receiving voice messages also), what do you think about that?

**Comment by innightwolfsleep** [comment]
siletto wasnt in my plans... but there is too much requests of thist integration. I will think about this.

**Comment by ch3shere** [comment]
That would be awesome. I tried to play it out with api cals, but not
getting results

**Comment by ch3shere** [comment]
Just a hint:

Siletto generated answers stored at: `<audio src="file/extensions/silero_tts/outputs/None_1685360029.wav" controls autoplay></audio>`

so we just need the bot to attach this file as answer message

**Comment by innightwolfsleep** [comment]
SileroTTS added. 
Cant reuse silero stright from ooba, so it is not need to acticvate silero extension.
